### PR TITLE
Squeeze image while scaling

### PR DESCRIPTION
**Pull request by dan-lee**
This PR contains
* Update React
* 'Squeeze' the image into the boundaries when scaling and rescaling the image. Not perfect yet but better than massive whitespace.
* Some code clean up
* ES6 syntax changes (thanks to `reactify`)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -87,41 +118,43 @@
...
+
+            if (isTouchDevice) React.initializeTouchEvents(true);
         },
 
-        componentWillUnmount: function () {
-            document && document.removeEventListener('mousemove', this.handleMouseMove, false);
-            document && document.removeEventListener('mouseup', this.handleMouseUp, false);
+        componentWillUnmount() {
+            document.removeEventListener(deviceEvents.native.move, this.handleMouseMove, false);
+            document.removeEventListener(deviceEvents.native.up, this.handleMouseUp, false);
```
Comment (mosch) [inline index.js]: The missing "document" was causing problems in nodejs environment. Removed this on purpose?

**Comment by dan-lee** [comment]
Everything worked fine without it and I didn't see the necessity to keep it.
What do you mean with problems in nodejs environment? This should run in the browser, right?

File: index.js (hunk 1 above)
Comment (dan-lee) [inline index.js]: Everything worked fine without it and I didn't see the necessity to keep it.
What do you mean with problems in nodejs environment? This should run in the browser, right?

Comment (mosch) [inline index.js]: For isomorphic apps - webapps that get prerendered on the server side. They must run on server-side nodejs and in browser! Check out https://github.com/mosch/react-avatar-editor/issues/17 :+1:

Comment (dan-lee) [inline index.js]: I see, didn't know about that. I'll put it back in. :+1:

**Pull request merged**